
package baidu.bfs;

message FileInfo {
    optional int64 entry_id = 1;
    optional int64 version = 2;
    optional int32 type = 3 [default = 0755];
    repeated int64 blocks = 4;
    optional uint32 ctime = 5;
    optional string name = 6;
    optional int64 size = 7;
    optional int32 replicas = 8;
    optional int64 parent_entry_id = 9;
    optional int32 owner = 10;
}

message ChunkServerInfo {
    optional int32 id = 1;
    optional string address = 2;
    optional int32 last_heartbeat = 3;
    optional int64 data_size = 4;
    optional int64 disk_quota = 5;
    optional int32 block_num = 7;
    optional bool is_dead = 8;
    optional int32 status = 9;
    optional int32 ip = 10;
    optional int32 buffers = 11;
    // running status
    optional int32 w_qps = 21;
    optional int64 w_speed = 22;
    optional int32 r_qps = 23;
    optional int64 r_speed = 24;
    optional int64 writing_buffers = 25;
    optional int64 active_blocks = 26;
}

